src/main/webapp/components/query-editor/query-editor.tsx

```diff
@@ -1,8 +1,45 @@
 import Box from '@material-ui/core/Box'
 import Button from '@material-ui/core/Button'
+import Divider from '@material-ui/core/Divider'
+import IconButton from '@material-ui/core/IconButton'
+import Menu from '@material-ui/core/Menu'
+import TextField from '@material-ui/core/TextField'
+import MoreVertIcon from '@material-ui/icons/MoreVert'
+import { set } from 'immutable'
 import React from 'react'
+import useAnchorEl from '../../react-hooks/use-anchor-el'
 import { AttributeDefinition, QueryType } from '../query-builder/types'
 
+const Header = (props: any) => {
+  const [anchorEl, handleOpen, handleClose, open] = useAnchorEl(props)
+  return (
+    <Box display="flex" style={{ padding: 8 }} alignItems="center">
+      <TextField
+        fullWidth
+        value={props.query.title}
+        variant="outlined"
+        label="Search Title"
+        onChange={event => {
+          props.onChange(set(props.query, 'title', event.target.value))
+        }}
+        autoFocus
+      />
+      <div style={{ marginLeft: 10 }} ref={props.addOptionsRef} />
+      <IconButton
```

Comment:
Should we hide the icon button if there are no query interactions?

Reply:
maybe we could do this if we bring this query editor and the search form query editor in sync